mesonbuild/backend/xcodebackend.py

```diff
@@ -53,6 +53,37 @@
                 }
 BOOL2XCODEBOOL = {True: 'YES', False: 'NO'}
 LINKABLE_EXTENSIONS = {'.o', '.a', '.obj', '.so', '.dylib'}
+XCODEVERSIONS = {'1500': ('Xcode 15.0', 60),
+                 '1400': ('Xcode 14.0', 56),
+                 '1300': ('Xcode 13.0', 55),
+                 '1200': ('Xcode 12.0', 54),
+                 '1140': ('Xcode 11.4', 53),
+                 '1100': ('Xcode 11.0', 52),
+                 '1000': ('Xcode 10.0', 51),
+                 '930': ('Xcode 9.3', 50),
+                 '800': ('Xcode 8.0', 48),
+                 '630': ('Xcode 6.3', 47),
+                 '320': ('Xcode 3.2', 46),
+                 '310': ('Xcode 3.1', 45)
+                 }
+
+# Detect what version of Xcode is installed on the machine.
+def autodetect_xcode_version():
+    try:
+        pc = subprocess.Popen(['xcodebuild', '-version'], encoding='utf-8', stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    except FileNotFoundError:
+        raise MesonException('Could not detect Xcode. Please install it if you wish to use the Xcode backend.')
+    (stdout, stderr) = pc.communicate()
+    if stderr:
+        raise MesonException(f'An error occurred while detecting Xcode: {stderr}')
+    version = int(re.search(r'\d*\.\d*\.*\d*', stdout).group(0).split('.'))
+    # If the version number does not have two decimal points, pretend it does.
+    if stdout.count('.') < 2:
+        version *= 10
+    for v, _ in XCODEVERSIONS.items():
+        if int(v) <= version:
+            return XCODEVERSIONS[v]
```

Comment:
```suggestion
    for v, r in XCODEVERSIONS.items():
        if int(v) <= version:
            return r
```
Since you're already iterating the returned value anyway